### ISSUE TITLE
PSR2 SwitchDeclaration sniff doesn't detect, or correctly fix, case body on same line as statement

### ISSUE THREAD

**Issue by oujesky** [squizlabs/PHP_CodeSniffer#621]
When running `PSR2.ControlStructures.SwitchDeclaration` sniff fix on following code:

```php
switch ($foo) {
    case 1: $bar = 1; break;
    case 2: $bar = 2; break;
    case 21:
    case 3: return 3;
    default: $bar = 0;
}
```

the expected result should be 

```php
switch ($foo) {
    case 1:
        $bar = 1;
        break;
    case 2:
        $bar = 2;
        break;
	case 21:
	case 3:
	    return 3;
	default:
	    $bar = 0;
}
```

but the actual result is messed up:

```php
switch ($foo) {
    case 1: $barbreak;
    case 2: $barbreak;
    case 21:
    case 3return 3;
    default: $bar = 0;
}
```

It seems the last few tokens before `case` end (be it `break` or `return`) are just removed. 

I can supply this code to `SwitchDeclarationUnitTest.inc` and `SwitchDeclarationUnitTest.inc.fixed` respectively, however I am not sure I can come with actual PR to fix this.

**Comment by gsherwood** [comment]
This is what the latest stable PHPCBF is giving me:
```php
switch ($foo) {
    case 1: $bar = 1;
        break;
    case 2: $bar = 2;
        break;
    case 21:
    case 3:
        return 3;
    default: $bar = 0;
}
```
I don't have any removed characters at all, although no newline is inserted after the CASE statements because PHPCS isn't checking for that in PSR2, which I think it should be.

Is this the only code in the file you are checking? Which version are you using?

**Comment by oujesky** [comment]
I can reproduce this issue with latest stable version

```
c:\phpcs>php phpcs.phar --version
PHP_CodeSniffer version 2.3.2 (stable) by Squiz (http://www.squiz.net)
```

with following code (saved as `test.php` in the next listings):

```php
<?php

switch ($foo) {
    case 1: $bar = 1; break;
    case 2: $bar = 2; break;
    case 21:
    case 3: return 3;
    default: $bar = 0;
}

```

```
c:\phpcs>php phpcs.phar --standard=PSR2 --sniffs=PSR2.ControlStructures.SwitchDeclaration test.php

FILE: C:\phpcs\test.php
----------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 3 LINES
----------------------------------------------------------------------
 4 | ERROR | [x] Terminating statement must be indented to the same
   |       |     level as the CASE body
 5 | ERROR | [x] Terminating statement must be indented to the same
   |       |     level as the CASE body
 7 | ERROR | [x] Terminating statement must be indented to the same
   |       |     level as the CASE body
----------------------------------------------------------------------
PHPCBF CAN FIX THE 3 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 32ms; Memory: 1.75Mb
```

```
c:\phpcs>php phpcbf.phar --standard=PSR2 --sniffs=PSR2.ControlStructures.SwitchDeclaration test.php
Changing into directory C:\phpcs
Processing test.php [PHP => 72 tokens in 9 lines]... DONE in 9ms (3 fixable violations)
        => Fixing file: 0/3 violations remaining [made 7 passes]... DONE in 27ms
Patched 1 file
Time: 280ms; Memory: 1.75Mb
```

The result after runing PHPCBF:

```php
<?php

switch ($foo) {
    case 1: $barbreak;
    case 2: $barbreak;
    case 21:
    case 3return 3;
    default: $bar = 0;
}

```

I also reproduced this issue on current master running modified `SwitchDeclarationUnitTest` (you can check it in my fork: https://github.com/oujesky/PHP_CodeSniffer/commit/1c642ab884b9490639f361823cc49db1e26a59c7, test result: https://travis-ci.org/oujesky/PHP_CodeSniffer/builds/67162519)

```
--- CodeSniffer\Standards\PSR2\Tests\ControlStructures\SwitchDeclarationUnitTest.inc.fixed
+++ PHP_CodeSniffer
@@ -105,15 +105,9 @@
 }

 switch ($foo) {
-    case 1:
-        $bar = 1;
-        break;
-    case 2:
-        $bar = 2;
-        break;
+    case 1: $barbreak;
+    case 2: $barbreak;
     case 21:
-    case 3:
-        return 3;
-    default:
-        $bar = 0;
+    case 3return 3;
+    default: $bar = 0;
 }
```

**Comment by gsherwood** [comment]
Thanks, I can replicate it now. I was using the whole PSR2 standard and not limiting the check and fixes to just that single sniff.

**Comment by gsherwood** [comment]
I've committed a fix for this. That sample code now get fixed by this sniff to this:
```php
<?php
switch ($foo) {
    case 1: 
        $bar = 1;
        break;
    case 2: 
        $bar = 2;
        break;
    case 21:
    case 3:
        return 3;
    default: 
        $bar = 0;
}
```

Also works well with the other sniffs in the PSR2 standard.
